### ISSUE TITLE
Media not working via TinaCMS

### ISSUE THREAD

**Issue by alexjj** [danurbanowicz/halide#4]
I really like your site design, and the integration of TinaCMS and the image plugin is perfect! 💯

I've got the site setup from your instructions and it's deployed and TinaCMS works - I can edit config and add new projects, but the media part doesn't work and just gives `Error fetching media. Something went wrong while requesting the resource.`. I've been to tina cloud and done the sync media, but still no luck.

All I can see in the docs is setting the following in tina/config.js. 

```
  media: {
    tina: {
      mediaRoot: "assets/uploads",
      publicFolder: "",
    },
```

I wondered if publicFolder should be _site or something?

**Comment by alexjj** [comment]
Yeah I searched around and there wasn't anything clear. This is my first time with 11ty and TinaCMS so wasn't quite sure of either.

I also noticed that if I use TinaCMS to change the site metadata, it changes the `site_icon_svg` from `/assets/icons/favicon.svg` to `site_icon_svg: /assets/uploads/assets/icons/favicon.svg` which then causes the build to fail as there's nothing there. Perhaps related to the media issue.

I'll make sure it's the same version and see what else the console tells me.

Thanks.

**Comment by danurbanowicz** [comment]
I think this is a Content Security Policy (CSP) issue where Halide's default policy (configured in `netlify.toml`) is blocking Tina-hosted media sources.

Due to the Tina app's use of various resource domains, I am going to re-think Halide's default CSP behaviour. In the meantime you can do one of 2 things to get media in the CMS working again:-

- Set the CSP to report errors only by setting `reportOnly = true` in `netlify.toml`. Any CSP errors will show in the console but won't cause resources to be blocked
- Add `assets.tina.io ` to the `imgSrc` line in `netlify.toml`

You might need to add additional allowed resource URLs to your CSP if you are still seeing errors in the console.

Hope that helps for now!

**Comment by seasonedfish** [comment]
> * Set the CSP to report errors only by setting `reportOnly = true` in `netlify.toml`. Any CSP errors will show in the console but won't cause resources to be blocked
> * Add `assets.tina.io ` to the `imgSrc` line in `netlify.toml`

Had the same issue, and this worked! Thanks.

**Comment by danurbanowicz** [comment]
Hi @alexjj and @seasonedfish !

The CSP issue has since been fixed by #6 which disables the CSP completely in Tina CMS at `/admin/*`.

Not ideal but we have no control over where Tina CMS loads external resources from, so for now this is the best solution I can see. 😐

If you want to opt in to this change, you should bump your `netlify-plugin-csp-generator` version to the latest (currently 1.6.1) and add the following `exclude` line to your CSP config inside `netlify.toml`

```
[plugins.inputs]
    buildDir = "_site"
    exclude = ["admin/**/*"]
```

The plugin also has a useful new build debug property that will output exactly which paths are included by your CSP:

`debug = true`
